src/Mod/Draft/draftfunctions/move.py

```diff
@@ -215,4 +215,116 @@ def copy_moved_edge(object, edge_index, vector):
+
+        return shape
+
+    elif shape.ShapeType == "Solid":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        solid = Part.Solid(new_sub_shapes)
+        return solid
+
+    elif shape.ShapeType == "Shell":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        shell = Part.Shell(new_sub_shapes)
+        return shell
+
+    elif shape.ShapeType == "Face":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        face = Part.Face(new_sub_shapes)
+        return face
+
+    elif shape.ShapeType == "Wire":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        wire = Part.Wire(new_sub_shapes)
+        return wire
+
+    elif shape.ShapeType == "Edge":
+        # to be refined
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        edge = Part.makeLine(new_sub_shapes[0].Point, new_sub_shapes[1].Point)
+        return edge
+
+    elif shape.ShapeType == "Vertex":
+
+        for sv in selected_vertexes:
+            if shape.X == sv.X and shape.Y == sv.Y and shape.Z == sv.Z:
+                return Part.Vertex(App.Vector(shape.X + vector.x, shape.Y + vector.y, shape.Z + vector.z)) 
+        return shape
+
+    return shape
+
+    
+def parse_sub_shapes(shape, selected_vertexes, vector):
+    """ Parse a the subshapes of the given shape in order to
+    find modified shapes and substitute them to the originals.
+    """
+    print('Shape is a ' + shape.ShapeType + '\n')
+    sub_shapes = []
+    if shape.SubShapes:
+        for sub_shape in shape.SubShapes:
+            sub_shapes.append(parse_shape(sub_shape, selected_vertexes, vector))
+            
+    if len(sub_shapes) == 1:
+        return sub_shapes[0]
+    else:
+        return sub_shapes
+
+
+def moveSubElements(obj, sub_objects_names, vector):
+    """moveSubElements(obj, sub_objects_names, vector)
+    
+    Move the given object sub_objects according to a vector or crates an new one
+    if the object is not a Part::Feature.
+
+    Parameters
+    ----------
+    obj : the given object
+
+    sub_objects_names : list of subelement names
+        A list of subelement names to identify the subelements to move.
+
+    vector : Base.Vector
+        Delta Vector to move subobjects from the original position. 
+
+    Return
+    ----------
+    shape : Part.Shape
+        Return the new Shape or None.
+    """
+
+    import Part
+
+    shape = obj.Shape
+    new_shape = None
+
+    selected_vertexes = []
+    for sub_objects_name in sub_objects_names:
+        sub_object = obj.Shape.getElement(sub_objects_name)
+        selected_vertexes.extend(sub_object.Vertexes)
+
+    new_shape = parse_shape(shape, selected_vertexes, vector)
+    
+    if new_shape:
+        if hasattr(obj, 'TypeId') and obj.TypeId == 'Part::Feature':
+            obj.Shape = new_shape
+        else:
+            Part.show(new_shape)
```

Comment:
By the way, here, using `Part.show()` creates a new simple `Part::Feature`, but we don't save the object.

I suggest adding the object manually, with `addObject`.
```py
if hasattr(obj, 'TypeId') and obj.TypeId == 'Part::Feature':
    obj.Shape = new_shape
else:
    new_obj = App.ActiveDocument.addObject("Part::Feature", "Feature")
    new_obj.Shape = new_shape
```

The only difference, I guess, is that we are able to specify the `Name` of the new object. In general, I only use `Part.show` to quickly test code, but in actual production code, I think adding an explicit `Part::Feature` is better.